### PR TITLE
#124 alternate target search

### PR DESCRIPTION
**Pull request by amandhamija98**
This is an approach to solve issue 124. It uses the normal search box on the whatsapp web website to find contacts and groups that match the search query.
Currently this search method has been added as an alternative in case an exception arises from the current search method

### CODE REVIEW

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- wplay/messageblast.py  1
- wplay/wchat.py  1
- wplay/messagetimer.py  1
- wplay/onlinetracker.py  1
         

Clones added
============
- wplay/messageblast.py  1
- wplay/wchat.py  1
- wplay/messagetimer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5202116&bid=17063493)

File: wplay/utils/target_search.py [hunk 1]
```diff
@@ -55,12 +56,109 @@
...
+async def search_and_select_target_without_new_chat_button(page,target, hide_groups=False):
+    await __type_in_chat_or_message_search(page,target)
+    chats_messages_groups_elements_list = await __get_chats_messages_groups_elements(page)
+    contact_name_index_tuple_list = await __get_contacts_matched_with_query(chats_messages_groups_elements_list)
+    group_name_index_tuple_list = await __get_groups_matched_with_query(chats_messages_groups_elements_list,hide_groups)
+    target_tuple = (contact_name_index_tuple_list,group_name_index_tuple_list)
+    __print_target_tuple(target_tuple)
+    target_index_chosen = __ask_user_to_choose_the_filtered_target(target_tuple)
+
+    '''chosen_target will be a tuple (a,b) such that a is the name of the target and b is the
```
Comment (rpotter12) [inline wplay/utils/target_search.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5202116)

File: wplay/utils/target_search.py [hunk 2]
```diff
@@ -55,12 +56,109 @@
...
+    await __type_in_chat_or_message_search(page,target)
+    chats_messages_groups_elements_list = await __get_chats_messages_groups_elements(page)
+    contact_name_index_tuple_list = await __get_contacts_matched_with_query(chats_messages_groups_elements_list)
+    group_name_index_tuple_list = await __get_groups_matched_with_query(chats_messages_groups_elements_list,hide_groups)
+    target_tuple = (contact_name_index_tuple_list,group_name_index_tuple_list)
+    __print_target_tuple(target_tuple)
+    target_index_chosen = __ask_user_to_choose_the_filtered_target(target_tuple)
+
+    '''chosen_target will be a tuple (a,b) such that a is the name of the target and b is the 
+    index of that element in chats_messages_groups_elements_list'''
```
Comment (rpotter12) [inline wplay/utils/target_search.py]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5202116)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/wchat.py  1
- wplay/onlinetracker.py  1
- wplay/messageblast.py  1
- wplay/messagetimer.py  1
         

Clones added
============
- wplay/wchat.py  1
- wplay/messageblast.py  1
- wplay/messagetimer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5202116&bid=17063493)

**Comment by rpotter12** [comment]
@amandhamija98 Please resolve the conflicts

**Comment by amandhamija98** [comment]
Sorry I am not able to understand which conflicts.
Are you referring to things like clones added?

**Comment by amandhamija98** [comment]
Sure, I'll just do that

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/messagetimer.py  1
- wplay/onlinetracker.py  1
- wplay/wchat.py  1
- wplay/messageblast.py  1
         

Clones added
============
- wplay/messagetimer.py  2
- wplay/wchat.py  1
- wplay/messageblast.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5202116&bid=17063493)

**Comment by amandhamija98** [comment]
Thankyou!

**Pull request merged**